src/auth/src/retry.rs

```diff
@@ -64,37 +66,50 @@ impl Builder {
             None => Arc::new(Mutex::new(AdaptiveThrottler::default())),
         };
 
-        let retry_policy = match self.retry_policy {
+        let retry_policy: Arc<dyn RetryPolicy> = match self.retry_policy {
             Some(p) => p,
             None => AlwaysRetry.with_attempt_limit(1).into(),
         }
         .into();
 
-        TokenProviderWithRetry {
+        let (tx, rx) = watch::channel(None);
+
+        let refresher = TokenRefresher {
             inner: Arc::new(token_provider),
             retry_policy,
             backoff_policy,
             retry_throttler,
-        }
+            token_watch_tx: tx,
+            last_delay: Arc::new(Mutex::new(None)),
+        };
+
+        tokio::spawn(refresher.run());
```

Comment:
This background task is going to stop once a token is obtained successfully I believe? What happens when we need the next token then?